src/router.cls.php

```diff
@@ -501,12 +506,30 @@ private function verify_action()
 		// Each action must have a valid nonce unless its from admin ip and is public action
 		// Validate requests nonce (from admin logged in page or cli)
 		if (!$this->verify_nonce($action)) {
-			// check if it is from admin ip
-			if (!$this->is_admin_ip()) {
+			// check if action is from admin ip. skip test for action Core::ACTION_QS_PURGE.
+			if ( $action != Core::ACTION_QS_PURGE && !$this->is_admin_ip()) {
 				Debug2::debug('[Router] LSCWP_CTRL query string - did not match admin IP: ' . $action);
 				return;
 			}
 
+			$save_blog = get_current_blog_id();
+			if ($_REQUEST['switch_blog']) {
+				// If request parameter "switch_blog", switch to correct blog to generate hash.
+				switch_to_blog($_REQUEST['switch_blog']);
+			}
+			$hash = Router::get_hash(self::VALIDATE_PURGE);
+			if ($_REQUEST['switch_blog']) {
+				// Restore blog if needed.
+				switch_to_blog($save_blog);
```

Comment:
This switch back and forth is a bit complicated. Should store the hash in the target blog. Please switch at https://github.com/litespeedtech/lscache_wp/pull/690/files#diff-7efe974c01bd97f4667ac581f8ce7fdf45b700734b28b2c945059af3e4b4059aR160